File: Supplemental/Java Solution/Option.java
public class Option
{

	public String optionID;
	public String pollID;
	public String name;
	public int maxSize;

	public Option(String optionID, String pollID, String name, int maxSize)
	{
		this.optionID = optionID;
		this.pollID = pollID;
		this.name = name;
		this.maxSize = maxSize;
	}

	@Override
	public boolean equals(Object obj)
	{
		if (obj == null)
			return false;
		if (obj instanceof Option)
		{
			return optionID.equals(((Option) obj).optionID)
					&& pollID.equals(((Option) obj).pollID)
					&& name.equals(((Option) obj).name)
					&& maxSize == ((Option) obj).maxSize;
		}
		return false;
	}

	@Override
	public int hashCode()
	{
		return optionID.hashCode() * pollID.hashCode() * name.hashCode()
				* (maxSize +1);
	}
}

File: Application/Entities/Option.php
<?php
namespace Application\Entities;

class Option extends \Spot\Entity
{
    protected static $_datasource = 'options';

    public static function fields()
    {
        return array(
            'id' => array('type' => 'int', 'primary' => true, 'serial' => true),
            'pollId' => array('type' => 'int'),
            'name' => array('type' => 'string'),
            'maxSize' => array('type' => 'int'),
        );
    }
    
    public static function relations()
    {
        return array(
            'answers' => array(
                'type' => 'HasMany',
                'entity' => '\Application\Entities\Answer',
                'where' => array('optionId' => ':entity.id'),
                'order' => array('id' => 'ASC')
            )
        );
    }
}
